internal/proxy/proxy_config.go

```diff
@@ -48,6 +48,7 @@ type UpstreamConfig struct {
 
 	SkipAuthCompiledRegex []*regexp.Regexp
 	AllowedGroups         []string
+	TLSVerify             bool
```

Comment:
It looks like in this case, the default for `TLSVerify` would be false,  which would be a less secure default. Can you change this field to be `InsecureSkipVerify`, which when set to true, will enable `InsecureSkipVerify`? This will allow us to default to a more secure configuration.

Reply:
Good point, and agree that default should be to verify. I meant to address that and forgot. However `InsecureSkipVerify` is a good name only inside a config struct named `TLSConfig`. It's not obvious that it's even TLS related outside that context. Given that, I'd be inclined to explicitly set a default rather than relying on zero values.

Reply:
What if we make the name `TLSSkipVerify`?

Reply:
Main reason I'd choose `TLSVerify` over `TLSSkipVerify` is that `TLSVerify` / `TLS_VERIFY` and similar are in wide usage in other tools, whereas `TLSSkipVerify` would be relatively uncommon. I also generally don't like to let language details (like what Go's zero values are) to leak out into user-facing interface design, and I think of config as part of the interface design.

Reply:
That said, there's certainly other tools using `TLSSkipVerify`, like Vault and KSonnet.